forc-pkg/src/manifest/dep_modifier.rs

```
@@ -0,0 +1,1133 @@
+            entry = "main.sw"
+            license = "Apache-2.0"
+            name = "package-1"
+
+            [dependencies]
+            foo = "1.0.0"
+            bar = "2.0.0"
+        "#;
+
+        let mut doc: DocumentMut = toml_str.parse().expect("failed to parse TOML");
+
+        remove_deps_from_table(&mut doc, "dependencies", &["foo"])?;
+
+        let table = doc["dependencies"].as_table().unwrap();
+        assert!(!table.contains_key("foo"));
+        assert!(table.contains_key("bar"));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_remove_multiple_deps() -> Result<()> {
+        let toml_str = r#"
+            [project]
+            authors = ["Fuel Labs <contact@fuel.sh>"]
+            entry = "main.sw"
+            license = "Apache-2.0"
+            name = "package-1"
+
+            [dependencies]
+            foo = "1.0.0"
+            bar = "2.0.0"
+            baz = "3.0.0"
+        "#;
+
+        let mut doc: DocumentMut = toml_str.parse().expect("failed to parse TOML");
+        remove_deps_from_table(&mut doc, "dependencies", &["foo", "bar"])?;
+
+        let table = doc["dependencies"].as_table().unwrap();
+        assert!(!table.contains_key("foo"));
+        assert!(!table.contains_key("bar"));
+        assert!(table.contains_key("baz"));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_remove_from_missing_section() {
+        let toml_str = r#"
+            [project]
+            authors = ["Fuel Labs <contact@fuel.sh>"]
+            entry = "main.sw"
+            license = "Apache-2.0"
+            name = "package-1"
+
+            [dependencies]
+            foo = "1.0.0"
+        "#;
+
+        let mut doc: DocumentMut = toml_str.parse().expect("failed to parse TOML");
+
+        let result = remove_deps_from_table(&mut doc, "contract-dependencies", &["foo"]);
+        assert!(result.is_err());
+        assert!(format!("{}", result.unwrap_err())
+            .contains("section [contract-dependencies] not found"));
```

hmm thinking about whether we should return the same error as if the dep is not found; rather than returning that the section does not exist; thoughts other contributors?
Wondering what cargo does.

---

yea cargo returns the following (example):
"error: the dependency `<dep>` could not be found in `dependencies`"

---

Oh okay will update that.